pyproject.toml

```diff
@@ -24,7 +24,7 @@ python = "^3.10"
 urllib3 = ">=2.3.0"
 ray = "2.44.0"
 semantic-version = "2.10.0"
-pyarrow = "*"
+pyarrow = "20.0.0"
```

Comment:
why not move to the latest version here? - https://pypi.org/project/pyarrow/21.0.0/

Reply:
There are some build errors for 21.0 with python 3.10. so we are proceeding with 20.0. 
  Add the installation prefix of "Arrow" to CMAKE_PREFIX_PATH or set
        "Arrow_DIR" to a directory containing one of the above files.  If "Arrow"
        provides a separate development package or SDK, be sure it has been
        installed.